bentoml/adapters/json_input.py

```diff
@@ -37,16 +37,15 @@ def __init__(self, is_batch_input=False, **base_kwargs):
         super(JsonInput, self).__init__(is_batch_input=is_batch_input, **base_kwargs)
 
     def handle_request(self, request: flask.Request, func):
-        if request.content_type == "application/json":
-            parsed_json = json.loads(request.get_data(as_text=True))
-        else:
+        if request.content_type != "application/json":
             raise BadInput(
                 "Request content-type must be 'application/json' for this "
                 "BentoService API"
             )
-
-        result = func(parsed_json)
```

Comment:
At least it should be func([parsed_json])

Reply:
This will be a regression right? Should we consider something like the `LegacyImageHandler`?

Reply:
LegacyJsonInput？